.github/workflows/packaging.yaml

```diff
@@ -0,0 +1,241 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: packaging
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "CMakeLists.txt"
+      - "src/nanoarrow/**"
+      - "r/**"
+      - "dev/release/**"
+      - ".github/workflows/packaging.yaml"
+
+  push:
+    # Automatically build on RC tags
+    branches-ignore:
+      - '**'
+    tags:
+      - 'apache-arrow-nanoarrow-*-rc*'
+
+concurrency:
+  group: ${{ github.repository }}-${{ github.ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  source:
+    name: Source
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Update tags
+        shell: bash
+        run: |
+          git fetch --tags --force origin
```

Comment:
This should be redundant as the checkout action will fetch tags for depth 0 (see [here](https://github.com/apache/arrow-nanoarrow/actions/runs/4057059415/jobs/6982341175#step:2:463))